src/history/errors.js

```diff
@@ -1,22 +1,67 @@
-export class NavigationDuplicated extends Error {
-  constructor (normalizedLocation) {
-    super()
-    this.name = this._name = 'NavigationDuplicated'
-    // passing the message to super() doesn't seem to work in the transpiled version
-    this.message = `Navigating to current location ("${
-      normalizedLocation.fullPath
-    }") is not allowed`
-    // add a stack property so services like Sentry can correctly display it
-    Object.defineProperty(this, 'stack', {
-      value: new Error().stack,
-      writable: true,
-      configurable: true
-    })
-    // we could also have used
-    // Error.captureStackTrace(this, this.constructor)
-    // but it only exists on node and chrome
-  }
+export const NavigationFailureType = {
+  NAVIGATION_GUARD_REDIRECT: 'NavigationRedirected',
+  NAVIGATION_ABORTED: 'NavigationAborted',
+  NAVIGATION_CANCELLED: 'NavigationCancelled',
+  NAVIGATION_DUPLICATED: 'NavigationDuplicated'
```

Comment:
The values do not matter much since they shouldn't be used directly but I think it's better to force that behavior by using numbers (like an enum), to prevent the mistake of people using a simple string comparison: `error.type === 'NavigationRedirected'` instead of `error.type === NavigationFailureType.redirected`

We should also keep the same keys as in https://github.com/vuejs/vue-router-next/blob/master/src/errors.ts#L27-L29

```suggestion
  redirected: 1,
  aborted: 2,
  cancelled: 3,
  duplicated: 4
```